src/include/migraphx/generate.hpp

```diff
@@ -36,19 +36,14 @@ inline namespace MIGRAPHX_INLINE_NS {
 template <class T, MIGRAPHX_REQUIRES(is_floating_point<T>{})>
 constexpr T normalize(unsigned long z)
 {
-    if(z == 0)
-        return T(0);
-    const auto max     = 32;
-    const double range = max / 2; // NOLINT
-    double result      = double(z % max) / range;
-    result -= 1;
-    return T(result);
+    const auto max = 1ULL << (sizeof(T) * 8 - 1);
+    return T(z % max);
 }
 
 template <class T, MIGRAPHX_REQUIRES(is_signed<T>{} and not is_floating_point<T>{})>
 constexpr T normalize(unsigned long z)
 {
-    const auto max      = 1ULL << (sizeof(T) * 5);
+    const auto max      = 1ULL << (sizeof(T) * 8 - 1);
```

Comment:
I dont think we want to change this for integers. It can create too large of integers which will introduce overflow.

Reply:
We are 1 bit less than the max number of bits.. should never overflow.. and the range is half of that ..

Reply:
We will overflow when its multiplied by any number larger than 2.

Reply:
(This is outdated code, BTW. We shouldn't overflow in this function.. !)